imgur.sh

```diff
@@ -89,14 +89,17 @@ while [ $# -gt 0 ]; do
 	url="${url%%</link>*}"
 	delete_hash="${response##*<deletehash>}"
 	delete_hash="${delete_hash%%</deletehash>*}"
-	echo $url
-	echo "Delete page: https://imgur.com/delete/$delete_hash" >&2
+	delete_page="https://imgur.com/delete/$delete_hash"
+	echo $url | sed 's/^http:/https:/'
+	echo "Delete page: $delete_page" >&2
 
 	# Append the URL to a string so we can put them all on the clipboard later
 	clip+="$url"
 	if [ $# -gt 0 ]; then
 		clip+=$'\n'
 	fi
+	# Keep a log of the uploads
+	echo "$(date +"%Y/%m/%d_%H:%M:%S")   $file   $url   $delete_page" >> $HOME/.imgur.sh.log
```

Comment:
- Let's wrap this in a conditional.
  
      if [ -n "${IMGUR_LOG_FILE-}" ]; then
  
  should do it. (That trailing dash is on purpose.)

- Then append it to that file. (Nitpick: I like no space between the `>>` and the file.)

- Please also use the ISO date format (`date -Isecond`), and use a tab characters between each pair of columns.

I'm not sure how valuable `$file` is, but short of checking whether it's stdin or a URL or a file and maybe running `realpath` on it, I don't know what to do. Probably just keep it simple and leave it how you are logging it.

Reply:
Conditional and date format done. I also re-ordered the variables to print $file at the end, as this makes the columns nicely justified instead of stair-cased.